sql/log.h

```diff
@@ -1129,7 +1132,7 @@ class MYSQL_BIN_LOG: public TC_LOG, private Event_log
   bool write_gtid_event(THD *thd, bool standalone, bool is_transactional,
                         uint64 commit_id,
                         bool commit_by_rotate,
-                        bool has_xid= false, bool ro_1pc= false);
+                        bool has_xid= false, bool ro_1pc= false, bool skip_all_events= false);
```

Comment:
`skip_all_events` I think is confusing terminology, as there is the option `--skip-annotate-row-events` which actually means to not write annotate row events into the binary log at all. Whereas this patch is writing the `skip_replicate` flag. I'd suggest a naming convention that relates to filtering (as that is the overall semantic) or `write_skip_rpl` (as that is the actual impact).

Also I think this line goes beyond 80 characters, so please split the line.